lewis/core/simulation.py

```diff
@@ -110,6 +112,38 @@ def _create_control_server(self, control_server):
                                         exclude=('add_adapter', 'remove_adapter', 'handle'))},
             control_server)
 
+    @property
+    def setups(self):
+        """
+        A list of setups that are available. Use :meth:`switch_setup` to
+        change the setup.
+        """
+        return list(self._device_builder.setups.keys()) if self._device_builder is not None else []
+
+    def switch_setup(self, new_setup):
+        """
+        This method switches the setup, which means that it replaces the currently
+        simulated device with a new device, as defined by the setup. This resets
+        the timers that keep track of the simulation.
+
+        If any error occurs during setup switching it is logged and re-raised.
+
+        :param new_setup: Name of the new setup to load.
+        """
+        try:
+            self._device = self._device_builder.create_device(new_setup)
+            self._reset_timers()
+            self._adapters.device = self._device
+        except Exception as e:
+            self.log.error(
+                'Caught an error while trying to switch setups. Setup not switched, '
+                'simulation continues: %s', e)
+            raise
+
+    def _reset_timers(self):
+        self._cycles = 0  # Number of cycles processed
+        self._runtime = 0.0  # Total simulation time processed
```

Comment:
Do we really want this to happen when changing setups?

These are properties of the simulation, rather than the device. Only the device has been re-created / reconfigured... Arguably, the simulation as a whole is carrying on uninterrupted, which means you could expect these counters to just continue as they were.